codec-native-quic/src/test/java/io/netty/incubator/codec/quic/QuicChannelConnectTest.java

```diff
@@ -729,6 +722,159 @@ public void testConnectWithoutTokenValidation(Executor executor) throws Throwabl
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("newSslTaskExecutors")
+    public void testKeyTypeChange(Executor executor) throws Throwable {
+        final CountDownLatch readLatch = new CountDownLatch(1);
+        Map<String, String> serverKeyTypes = new HashMap<>();
+        serverKeyTypes.put("RSA", "RSA");
+
+        Set<String> clientKeyTypes = new HashSet<>();
+        clientKeyTypes.add("RSA");
+
+        Channel server = QuicTestUtils.newServer(QuicTestUtils.newQuicServerBuilder(executor,
+                        QuicSslContextBuilder.forServer(
+                                        QuicTestUtils.SELF_SIGNED_CERTIFICATE.privateKey(), null,
+                                        QuicTestUtils.SELF_SIGNED_CERTIFICATE.certificate())
+                                .applicationProtocols(QuicTestUtils.PROTOS)
+                                .option(BoringSSLContextOption.SERVER_KEY_TYPES, serverKeyTypes)
+                                .earlyData(true)
+                                .build()),
+                TestQuicTokenHandler.INSTANCE, new ChannelInboundHandlerAdapter() {
+                    @Override
+                    public boolean isSharable() {
+                        return true;
+                    }
+                }, new ChannelInboundHandlerAdapter() {
```

Comment:
You want to use `ByteToMessageDecoder` as there is no guarantee that you will receive all 4 bytes at once.

Reply:
@normanmaurer